### PR TITLE
Follow vault list on my page

### PR DESCRIPTION
**Pull request by piotrkonowrocki**
# Follow vault list on my page

Covers:
https://app.shortcut.com/oazo-apps/story/5886/follow-vaults-overview-section-ui-component
https://app.shortcut.com/oazo-apps/story/5891/followed-vaults-list-view-vault
and a little bit more

**PLEASE READ through changes list, how to test and additional notes before testing.**

This PR adds skeleton preloader ONLY for follow vault. User's positions at this moment doesn't have preloaders at all and are still displayed in old layout - not in discovery table. This is for next PR as since entire follow vault feature is hidden behind feature flag, temporary inconstancy won't matter.

image
image
image
  
## Changes 👷‍♀️

- Huge Discover table overhaul allowing it to be more flexible and accepting more data than just discover,
- created api handler and overvable for loading user's followed vaults,
- created table with list of followed vaults for owner's page,
- created first version of skeleton preloader compontent to be used in other places - probably need to be expanded a little.
  
## How to test 🧪

- Go to owner's page (yours or somebody else's) and see "Follow" panel.
- see if preloader works as intended,
- see if empty state works as intended,
- see if list is populated with vaults that you decided to follow.

DO NOT report issues with following or unfollowing, this PR is only about list on owner's page.

## Additional notes and know bugs 🗒️

- You won't be able to unfollow vault until is merged: https://github.com/OasisDEX/oasis-borrow/pull/1861 is merged,
- sometimes during loading there is random circle preloaded on skeleton - this is a bug from component below with product cards and will be addressed with adding skeleton preloaders to the rest of that page,
- some header labels in table might be different than in design, but the names were unified to the ones from Discover - if they are going to be changed it's a matter of different discussion,
- guni vault shows incorrect data for PnL and liquidity, last 7 days yield is unavailable at all - this is outside of scope for this PR.

### CODE REVIEW

**Comment by johnnieskywalker** [comment]
image

image

followed vaults are displayed correctly 

image

**Comment by johnnieskywalker** [comment]
Only positions for currently selected network are displayed 
image

view position works as expected

File: features/discover/common/DiscoverError.tsx [hunk 1]
```diff
@@ -16,7 +22,9 @@ export function DiscoverError({ error }: { error?: DiscoverDataResponseError })
         color: 'neutral80',
       }}
     >
-      {t(`discover.api-error.${error?.code}`)}
+      {message}
+      {message && error && '<br />'}
+      {error && t(`discover.api-error.${error?.code}`)}
```
Comment (piekczyk) [inline features/discover/common/DiscoverError.tsx:27]: `error.code`

File: features/discover/common/DiscoverResponsiveTable.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  rows: DiscoverTableRowData[]
+  skip?: string[]
+  tooltips?: string[]
+  onBannerClick?: (link: string) => void
+  onPositionClick?: (cdpId: string) => void
+}) {
+  const isSmallerScreen = useMediaQuery(`(max-width: ${theme.breakpoints[2]})`)
+
+  return (
+    <>
```
Comment (piekczyk) [inline features/discover/common/DiscoverResponsiveTable.tsx:33]: Do we need fragment here? Most likely you could just return condition directly.

File: components/AppContext.ts [hunk 3]
```diff
@@ -1346,6 +1348,28 @@ export function setupAppContext() {
...
+
+  const followedList$ = memoize(
+    curry(createMakerPositionsList$)(
+      context$,
+      ilksWithBalance$,
+      memoize(
+        curry(vaultsWithHistory$)(
+          chainContext$,
+          curry(decorateVaultsWithValue$)(vaultsFromId$, exchangeQuote$, userSettings$),
+          1000 * 60,
```
Comment (johnnieskywalker) [inline components/AppContext.ts:1367]: please name the variable `refreshInterval`

**Review by johnnieskywalker — CHANGES_REQUESTED** [review]

File: components/AppContext.ts (hunk 3 above)
Comment (piotrkonowrocki) [reply components/AppContext.ts]: Fixed.

**Review by piekczyk — CHANGES_REQUESTED** [review]

File: features/discover/common/DiscoverResponsiveTable.tsx (hunk 2 above)
Comment (piotrkonowrocki) [reply features/discover/common/DiscoverResponsiveTable.tsx]: And fixed.

File: components/Skeleton.tsx [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+      }}
+    />
+  )
+}
+
+export function Skeleton({ lines = 1, gap = 3, ...rest }: SkeletonProps) {
+  return (
+    <Grid gap={gap}>
+      {[...Array(lines)].map(() => (
+        <SkeletonLine {...rest} />
```
[the remark this answers is not in the document]
Comment (piotrkonowrocki) [reply components/Skeleton.tsx]: Fixed.

File: blockchain/vaults.ts [hunk 5]
```diff
@@ -81,6 +81,36 @@ export function createVaults$(
   )
 }
 
+export function createVaultsFromIds$(
+  refreshInterval: Observable<any>,
```
[the remark this answers is not in the document]
Comment (piotrkonowrocki) [reply blockchain/vaults.ts]: Fixed.

**Review by johnnieskywalker — APPROVED** [review]

**Review by piekczyk — APPROVED** [review]

**Pull request merged**